sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/uninstall/UninstallRecorder.java

```diff
@@ -51,46 +42,59 @@ public void record(OfferRecommendation offerRecommendation) throws Exception {
         List<Protos.TaskInfo> tasksToUpdate = stateStore.fetchTasks().stream()
                 .filter(taskSpec -> containsResource(taskSpec, resource))
                 .collect(Collectors.toList());
-        logger.info("Resource {} found in {} task(s).", resource.getName(), tasksToUpdate.size());
-        tasksToUpdate.forEach(taskInfo ->
-                logger.info("Resource {} found in task: {}",
-                        resource.getName(),
-                        TextFormat.shortDebugString(taskInfo)));
-        if (!tasksToUpdate.isEmpty()) {
-            stateStore.storeTasks(updateResources(resource, tasksToUpdate));
-
-            // broadcast uninstallRecommendation to each UninstallStep in resource phase
-            List<OfferRecommendation> uninstallRecommendations = Collections.singletonList(uninstallRecommendation);
-            resourceSteps.forEach(step -> step.updateOfferStatus(uninstallRecommendations));
+        if (tasksToUpdate.isEmpty()) {
+            return;
         }
+
```

Comment:
Seems like a lot of code duplication here with DecommissionRecorder. Can we factor that out into a base class?

Reply:
There is sort of code duplication, but it's not really worth trying to factor out due to the subtle differences between these two.